### PR TITLE
Make X-Forwarded-For handling configurable

### PR DESCRIPTION
**Pull request by mythmon**
Fixes bug 1268214.

@Osmose r?

### CODE REVIEW

File: normandy/recipes/tests/test_models.py [hunk 1]
```diff
@@ -208,20 +208,47 @@ def test_matches_deals_with_none(self):
 
 
 class TestClient(object):
-    @pytest.mark.parametrize('x_forwarded_for,expected_ip', [
-        ('192.0.2.0,127.0.0.1', '192.0.2.0'),
-        ('192.0.2.0', '192.0.2.0'),
-    ])
-    def test_country_x_forwarded_for(self, rf, x_forwarded_for, expected_ip):
-        client = Client(rf.get('/', HTTP_X_FORWARDED_FOR=x_forwarded_for))
+    def test_country_remote_addr_fallback(self, rf):
```
Comment (Osmose) [inline normandy/recipes/tests/test_models.py]: This test isn't necessary anymore, you're testing the same thing as when there are no proxies.

File: normandy/recipes/tests/test_models.py [hunk 2]
```diff
@@ -208,20 +208,47 @@ def test_matches_deals_with_none(self):
...
+        forwarded_for = ', '.join(['fake', client_ip])
+        req = rf.get('/', HTTP_X_FORWARDED_FOR=forwarded_for, REMOTE_ADDR=nginx_ip)
+        client = Client(req)
+
+        with patch('normandy.recipes.models.get_country_code') as get_country_code:
+            assert client.country == get_country_code.return_value
+            get_country_code.assert_called_with(client_ip)
+
+    def test_x_forwarded_for_two_proxies(self, rf, settings):
+        """If there is one proxy, the right-most value in HTTP_X_FORWARDED_FOR should be used."""
```
Comment (Osmose) [inline normandy/recipes/tests/test_models.py]: You forgot to update this docstring.

File: normandy/recipes/models.py [hunk 3]
```diff
@@ -293,10 +294,15 @@ def locale(self):
 
     @cached_property
     def country(self):
-        try:
-            ip_address = self.request.META['HTTP_X_FORWARDED_FOR'].split(',')[0]
-        except (KeyError, IndexError):
+        if settings.NUM_PROXIES == 0:
```
Comment (Osmose) [inline normandy/recipes/models.py]: This might fit better in a utility method, like `get_client_ip(request)` or something.

It'd be easier to test too.

**Comment by Osmose** [comment]
r+wc

**Comment by mythmon** [comment]
@Osmose Can you look over this again?

File: normandy/base/utils.py [hunk 4]
```diff
@@ -26,3 +27,15 @@ def urlparams(url, fragment=None, **kwargs):
         urlencode(query, doseq=True),
         fragment if fragment is not None else parsed.fragment
     ))
+
+
+def get_client_ip(request):
```
Comment (Osmose) [inline normandy/base/utils.py]: A docstring would be swell.

**Comment by Osmose** [comment]
lgtm, r+!

**Pull request merged**